moveit_ros/moveit_servo/include/moveit_servo/datatypes.hpp

```diff
@@ -36,22 +31,52 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *******************************************************************************/
 
+/*      Title     : datatypes.hpp
+ *      Project   : moveit_servo
+ *      Created   : 05/06/2023
+ *      Author    : V Mohammed Ibrahim
+ */
+
 #pragma once
 
-#include <memory>
-#include <boost/pool/pool_alloc.hpp>
+namespace moveit_servo
+{
+// Datatypes used by servo
+
+typedef Eigen::VectorXd JointJog;
 
-namespace moveit
+struct Pose
 {
-namespace util
+  std::string frame_id;
+  Eigen::Isometry3d pose;
+};
+
+struct Twist
 {
-// Useful template for creating messages from a message pool
-template <typename T>
-std::shared_ptr<T> make_shared_from_pool()
+  std::string frame_id;
+  Eigen::Vector<double, 6> velocities;
+};
+
+typedef std::variant<JointJog, Twist, Pose> ServoInput;
+
+struct KinematicState
 {
-  using allocator_t = boost::fast_pool_allocator<std::shared_ptr<T>>;
-  return std::allocate_shared<T, allocator_t>(allocator_t());
-}
+  std::vector<std::string> joint_names;
+  std::vector<double> positions, velocities, accelerations;
 
-}  // namespace util
-}  // namespace moveit
+  KinematicState(const int num_joints)
+  {
+    joint_names.resize(num_joints);
+    positions.resize(num_joints);
+    velocities.resize(num_joints);
+    accelerations.resize(num_joints);
+  }
+};
+
+enum class CommandType
+{
+  JOINT_JOG = 0,
+  TWIST,
+  POSE
```

Comment:
Might as well initialize all of them

```suggestion
  JOINT_JOG = 0,
  TWIST = 1,
  POSE = 2
```